src/storage/storage.cc

```diff
@@ -235,19 +235,20 @@ Status Storage::Open(bool read_only) {
   db_closing_ = false;
 
   bool cache_index_and_filter_blocks = config_->rocks_db.cache_index_and_filter_blocks;
+  size_t block_cache_size = config_->rocks_db.block_cache_size * MiB;
   size_t metadata_block_cache_size = config_->rocks_db.metadata_block_cache_size * MiB;
   size_t subkey_block_cache_size = config_->rocks_db.subkey_block_cache_size * MiB;
+  if (block_cache_size == 0) {
+    block_cache_size = metadata_block_cache_size + subkey_block_cache_size;
+  }
 
   rocksdb::Options options = InitRocksDBOptions();
   if (auto s = CreateColumnFamilies(options); !s.IsOK()) {
     return s.Prefixed("failed to create column families");
   }
 
   std::shared_ptr<rocksdb::Cache> shared_block_cache;
-  if (config_->rocks_db.share_metadata_and_subkey_block_cache) {
-    size_t shared_block_cache_size = metadata_block_cache_size + subkey_block_cache_size;
-    shared_block_cache = rocksdb::NewLRUCache(shared_block_cache_size, -1, false, 0.75);
-  }
+  shared_block_cache = rocksdb::NewLRUCache(block_cache_size, -1, false, 0.75);
 
   rocksdb::BlockBasedTableOptions metadata_table_opts = InitTableOptions();
   metadata_table_opts.block_cache =
```

Comment:
Can use shared_block_cache directly, also other column family table's block cache options can remove the NewLRU condition.